### PR TITLE
Try to enable OSX build

### PR DESCRIPTION
**Pull request by rcurtin**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

So, I only sort of know what I am doing, but I think that mlpack isn't built for OS X, and maybe the change I made here might make that happen.  Let's find out!

### CODE REVIEW

**Comment by rcurtin** [comment]
@conda-forge-admin, please rerender

**Comment by rcurtin** [comment]
@conda-forge-admin, please rerender

**Comment by rcurtin** [comment]
This needs a port of https://github.com/mlpack/mlpack/pull/3297; once that PR is merged, I'll port the patch here and this should succeed.

**Comment by coatless** [comment]
@rcurtin maybe we should disable the `linux` & `windows` builds to speed up the build time on `mac`?

**Comment by rcurtin** [comment]
Oh, good idea!  I forgot that I could do that.  In my next hapless commit I'll make that change too. :+1:  (If you really want to see a trail of sadness see the wreckage at https://github.com/mlpack/mlpack-wheels ...)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,12 +9,15 @@ source:
...
     - python_copy_target_fix.patch
-    - setup_py_msvc.patch
     - win_py_install_prefix.patch
+    - osx_cmake.patch
+    - python_flags.patch
+    - osx_link_m.patch
 
 build:
-  number: 3
-  skip: true  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml:17]: Could do something like this and re-render to disable the other builds

```suggestion
  skip: true  # [win or linux]
```

Comment (rcurtin) [reply recipe/meta.yaml:17]: Thanks, fix applied.  Hope you don't mind the emails for each of my (likely many) subsequent attempts.

**Comment by rcurtin** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:17]: Ofc. No worries. Get many notifications from conda-forge independent of activity in any one PR :)

File: recipe/build.sh [hunk 2]
```diff
@@ -44,6 +44,7 @@ ${PYTHON} --version
 echo ""
 echo "show setuptools version"
 echo ""
+conda install -c anaconda pip
```
Comment (jakirkham) [inline recipe/build.sh]: This should be added in `requirements/host`

https://github.com/conda-forge/mlpack-feedstock/blob/ff1ff4f873d4c3fe5fadba59a0523a02a699dd3b/recipe/meta.yaml#L28

Comment (rcurtin) [reply recipe/build.sh]: Thanks---this was just screwing around, I think I misunderstood how the build environment works.  I might have a solution now... not clear yet...

**Comment by rcurtin** [comment]
@conda-forge-admin, please rerender

**Comment by rcurtin** [comment]
Sweet, I got it to work!  I didn't expect OS X to give so much trouble.  I have one more cleanup commit to do, and if that build succeeds and nobody has any other comments, I'll go ahead and merge. :rocket:

**Comment by jakirkham** [comment]
Hooray! 🥳 Thanks Ryan 🙏

**Pull request merged**